### PR TITLE
use docker image 0.5.56

### PR DESCRIPTION
**Pull request by pankore**
Problem
* Move forward to the latest docker images.

Change overview
* Change all images to 0.5.56

Testing
CI will validate this

### CODE REVIEW

**Comment by pankore** [comment]
The prerequisite of this PR is #15252

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
Apparently image is not yet built. I will start  a build now (will take about 30 min)

**Comment by andy31415** [comment]
Apparently only 0.5.54 is in master.

**Comment by andy31415** [comment]
Merged #15252 and now started a build for 0.5.55. Expect it to take 30min-1hr.

**Comment by pankore** [comment]
Thank you!!

**Comment by andy31415** [comment]
@pankore - Ameba build failure (about pwrpc) seems a real failure - retried it several times and Amebad generally is not flaky.

**Comment by pankore** [comment]
> @pankore - Ameba build failure (about pwrpc) seems a real failure - retried it several times and Amebad generally is not flaky.

Noted, we will check on that. Thanks!

**Comment by pankore** [comment]
The Build example error is expected to be fixed by #15388 .

**Comment by pankore** [comment]
Hi @andy31415 @woody-apple ,
Please help to check this PR again, the build example error has been fixed.
Thanks

**Comment by pankore** [comment]
Hi reviewer [andy31415](https://github.com/andy31415), [anush-apple](https://github.com/anush-apple), [balducci-apple](https://github.com/balducci-apple), [Byungjoo-Lee](https://github.com/Byungjoo-Lee), [bzbarsky-apple](https://github.com/bzbarsky-apple), [carol-apple](https://github.com/carol-apple), [cecille](https://github.com/cecille), [chrisdecenzo](https://github.com/chrisdecenzo), [chshu](https://github.com/chshu), [chulspro](https://github.com/chulspro), [Damian-Nordic](https://github.com/Damian-Nordic), [dhrishi](https://github.com/dhrishi), [electrocucaracha](https://github.com/electrocucaracha), [franck-apple](https://github.com/franck-apple), [gjc13](https://github.com/gjc13), [hawk248](https://github.com/hawk248), [holbrookt](https://github.com/holbrookt), [hrajendran-samsung](https://github.com/hrajendran-samsung), [isiu-apple](https://github.com/isiu-apple), [jelderton](https://github.com/jelderton), [jepenven-silabs](https://github.com/jepenven-silabs), [jmartinez-silabs](https://github.com/jmartinez-silabs), [kpschoedel](https://github.com/kpschoedel), [lazarkov](https://github.com/lazarkov), [LuDuda](https://github.com/LuDuda), [lzgrablic02](https://github.com/lzgrablic02),

Please help to check this PR as it is a blocking issue to PR #15257, many thanks!

**Review by andy31415 — APPROVED** [review]

File: scripts/build/builders/ameba.py [hunk 1]
```diff
@@ -51,6 +51,13 @@ def AppNamePrefix(self):
             raise Exception('Unknown app type: %r' % self)
 
 
+def OptionalFlags(self):
```
Comment (andy31415) [inline scripts/build/builders/ameba.py:54]: `self` is generally expected to be a member method. Please move as member method (call as self.OptionalFlags) and also fix naming - what is an optional flag? What is its use? what makes pigweed-app special?

From this context, I do not know what `-r` acutally means.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Changes requested for python code cleanlyness:

- `self` is generally reserved for instance methods not for free methods.
- the %s formatting will introduce extra spacing for non-pigweed apps

Try something like:
```py
cmd = '$AMEBA_PATH/project/realtek_amebaD_va0_example/GCC-RELEASE/build.sh '

if self.app.ExampleName == 'pigweed-app':
   # document here what the flag does and why we are adding it
   cmd += '-r '

# Document expected script arguments:
#   <build root> <build_system> <output_directory> <application>
cmd += ' '.join(self.root, 'ninja', self.output_dir, self.app.ExampleName)
```

File: scripts/build/testdata/build_all_except_host.txt [hunk 2]
```diff
@@ -2,13 +2,13 @@
 cd "{root}"
 
 # Generating ameba-amebad-all-clusters
-bash -c '$AMEBA_PATH/project/realtek_amebaD_va0_example/GCC-RELEASE/build.sh {root} ninja {out}/ameba-amebad-all-clusters all-clusters-app'
+bash -c '$AMEBA_PATH/project/realtek_amebaD_va0_example/GCC-RELEASE/build.sh  {root} ninja {out}/ameba-amebad-all-clusters all-clusters-app'
```
Comment (andy31415) [inline scripts/build/testdata/build_all_except_host.txt:5]: extra space should not be here - we try to make the `--dry-run` output be reasonably human-readable

File: scripts/build/builders/ameba.py (hunk 1 above)
Comment (pankore) [reply scripts/build/builders/ameba.py:54]: The optionalflag function is meant to send some flags to platform side for compilation.
The -r stands for rpc.

File: scripts/build/testdata/build_all_except_host.txt (hunk 2 above)
Comment (pankore) [reply scripts/build/testdata/build_all_except_host.txt:5]: Understood, this has been fixed.

**Review by andy31415 — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**